test/unit/storage_test.rb

```diff
@@ -138,12 +185,11 @@ def assert_sentinel_connector(client)
     assert_instance_of Redis::Client::Connector::Sentinel, connector
   end
 
-  def assert_client_config(client, host: nil, port: nil, url: nil, sentinels: nil)
-    raise "bad usage of #{__method__}" unless host || port || url
-    assert_equal client.port, port if port
-    assert_equal client.host, host if host
-    assert_equal client.options[:url], url if url
-    assert_equal client.options[:sentinels], sentinels if sentinels
+  def assert_client_config(client, opts)
+    raise "bad usage of #{__method__}" unless opts[:host] || opts[:port] || opts[:url]
```

Comment:
Not really needed now no? Maybe just assert it is not empty?

Reply:
Definitely *url* has to be checked for not empty.
Others, not needed.

Reply:
so if there are mandatory fields, we should make them mandatory in the method signature:
```ruby
def assert_client_config(client, url, opts)
```

Also, probably opts is not the best name for the hash containing the rest of assertions. That said, are you sure the client connection configuration object always has a valid `url` field?

Reply:
we are testing exactly that and, moreover, you removed *host, port* mode of redis connection configuration.

updated to read (and making sure it exists) *url*